### PR TITLE
add test to check whether tarsier artifacts still exist

### PR DESCRIPTION
**Pull request by seanmcguire12**
Test will be much more useful after colour tagging branch is merged.

### CODE REVIEW

File: tests/test_artifact_removal.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import os
+
+import pytest
+from playwright.async_api import async_playwright
+
+example_data = [
+    {
+        "file_name": "test_artifact_page.mhtml",
+        "artifact_selectors": [
+            "[__tarsier_id]"
```
Comment (asim-shrestha) [inline tests/test_artifact_removal.py:10]: does this ignore the starting text of the selector?

File: tests/test_artifact_removal.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        ],  # TODO: add more selectors once colour tagging is merged
+    },
+]
+
+
+@pytest.mark.parametrize("data", example_data)
```
Comment (asim-shrestha) [inline tests/test_artifact_removal.py:16]: let's parameterize by tarsier function as well

File: tests/test_artifact_removal.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    )
+
+    async with async_playwright() as p:
+        browser = await p.chromium.launch(headless=True)
+        page = await browser.new_page()
+
+        await page.goto(f"file://{html_file_path}")
+
+        _, _ = await tarsier.page_to_text(page, tag_text_elements=True)
+
```
Comment (asim-shrestha) [inline tests/test_artifact_removal.py:34]: should ideally test

- Run tarsier function without clean up
- Selectors exist
- Run tarsier cleanup
- Selectors gone

File: tests/test_artifact_removal.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    artifact_selectors = data["artifact_selectors"]
+
+    # Construct the path to the HTML file
+    html_file_path = os.path.abspath(
+        os.path.join(os.path.dirname(__file__), "mock_html", file_name)
+    )
+
+    async with async_playwright() as p:
+        browser = await p.chromium.launch(headless=True)
+        page = await browser.new_page()
```
Comment (asim-shrestha) [inline tests/test_artifact_removal.py:29]: We've written this stuff quite a bit. We can maybe wrap all of this stuff in a context manager function that takes filepath, and goes to the url and returns the page. Let's not do it in this PR but let's make a ticket

File: tests/test_artifact_removal.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+            "#__tarsier_id",
+        ],  # TODO: add more selectors once colour tagging is merged
+    },
+]
+
+tarsier_functions = [
+    (
+        "page_to_text",
+        {"tag_text_elements": True, "keep_tags_showing": True},
+        ["remove_tags"],
```
Comment (asim-shrestha) [inline tests/test_artifact_removal.py:19]: to make this more concrete and avoid dealing with strings, you can make these lambda functions that take in tarsier/the page as a param, and then you can call the functions with the desired params directly

File: tests/test_artifact_removal.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        # check if tarsier artifacts still exist
+        for selector in artifact_selectors:
+            elements = await page.query_selector_all(selector)
+            assert len(elements) > 0, (
+                f"Tarsier artifact '{selector}' not found in file: {file_name} "
+                f"after calling {function_name} with keep_tags_showing=True"
+            )
+
+        # run cleanup function(s)
+        for cleanup_function in cleanup_functions:
```
Comment (asim-shrestha) [inline tests/test_artifact_removal.py:59]: only one clean up function

File: tests/test_artifact_removal.py (hunk 5 above)
Comment (asim-shrestha) [reply tests/test_artifact_removal.py:19]: also avoids this get attribute stuff

File: tests/test_artifact_removal.py (hunk 4 above)
Comment (seanmcguire12) [reply tests/test_artifact_removal.py:43]: https://linear.app/reworkd/issue/APE-198/add-a-a-context-manager-function-for-tarsier-tests

File: tests/test_artifact_removal.py (hunk 6 above)
Comment (seanmcguire12) [reply tests/test_artifact_removal.py:59]: Will be more once we merge colour tagging

**Review by asim-shrestha — APPROVED** [review]

**Pull request merged**